File: leetcode/charcterReplacement.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        res = 0

        for c in set(s):
            cnt = 0
            left = 0
            for right in range(len(s)):
                if s[right] == c:
                    cnt += 1

                while (right - left + 1) - cnt > k:
                    if s[left] == c:
                        cnt -= 1
                    left += 1
                res = max(res, right - left + 1)
        return res

File: leetcode/charcterReplacement.go
func characterReplacement(s string, k int) int {
    var res int = 0
    for c := 'A'; c <= 'Z'; c++ {
        cnt := 0
        l := 0
        for r := 0; r < len(s); r++ {
            if s[r] == byte(c) {
                cnt++
            }

            for (r - l + 1) - cnt > k {
                if s[l] == byte(c) {
                    cnt--
                }
                l++
            }
            if res < r-l+1 {
                res = r-l+1
            }
        }    
    }
    return res
}
